store/tikv/gc_worker.go

```diff
@@ -43,10 +44,40 @@ type GCWorker struct {
 	lastFinish  time.Time
 	cancel      goctx.CancelFunc
 	done        chan error
+
+	session tidb.Session
+
+	spsession   tidb.Session
+	sessionLock sync.Mutex
+}
+
+// StartSafePointChecker triggers SafePoint Checker on each tidb
+func (w *GCWorker) StartSafePointChecker() {
+	w.spsession = createSession(w.store)
+	go func() {
+		for {
+			select {
+			case <-w.store.spMsg:
+				return
+			default:
+				spCachedTime := time.Now()
+				cachedSafePoint, err := w.loadSafePoint(gcSavedSafePoint)
+				if err == nil {
+					w.store.spMutex.Lock()
```

Comment:
I prefer to create an independent function in `store` to set `safePoint` and `spTime`